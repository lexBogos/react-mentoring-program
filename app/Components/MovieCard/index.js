import React from 'react';
import PropTypes from 'prop-types';
import './index.scss';

const MovieCard = ({movie, showMovieInfoPanel, onMovieChosen}) => 
    <div className='movieCard' >
        <img src={movie.movieUrl} width="240" height="320"/>
        <div>
            <h3>{movie.title}</h3>
            <span>{movie.genre}</span>
        </div>
        <span className='movieYear'>{movie.realiseDate}</span>
        <div className='movieCardActions'>
            <a href='#' onClick={(event) => {
                event.preventDefault();
                onMovieChosen(movie.id, 'EDIT_MOVIE');
                }}>edit</a>
            <a href='#' onClick={(event) => {
                event.preventDefault();
                onMovieChosen(movie.id, 'REMOVE_MOVIE');
                }}>delete</a>
            <a href='#' onClick={(event) => {
                event.preventDefault();
                showMovieInfoPanel(movie.id);
                window.scrollTo(0, 0);
                }}>...</a>    
        </div>
    </div>

MovieCard.propTypes = {
    movie: PropTypes.object.isRequired,
}

export default MovieCard;