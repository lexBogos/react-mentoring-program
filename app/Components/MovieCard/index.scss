.movieCard{
    height: 500px;
    width: 300px;
    border: 2px solid bisque;
    text-align: center;
    color:rgb(189,189,189);
}
.movieType{
    font-size: 16px;
    margin-right: 20px;
}
.movieYear{
    font-size: 20px;
    margin-right: 20px;
    border: 1px solid grey;
    border-radius: 0.1cm;
}