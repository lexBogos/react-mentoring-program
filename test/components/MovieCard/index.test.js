import React from 'react';
import MovieCard from '../../../app/Components/MovieCard/index';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Enzyme,{ shallow } from 'enzyme';
import { render, waitFor, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect';

Enzyme.configure({ adapter: new Adapter() });

let onMovieChosen;
let preventDefault;


const movie =  {
    title: 'Goodfellas',
    id: 'tt0099685',
    realiseDate: 1990,
    rate: 4.9
}


beforeEach(()=>{
    onMovieChosen = jest.fn(e=>e);
    preventDefault = jest.fn()
});

describe('MovieCard component', () => {
        it('render MovieCard component toMatchSnapshot', () => {
            const MovieCardObj = shallow(<MovieCard movie={movie} onMovieChosen={onMovieChosen}/>);
            
            expect(MovieCardObj.debug()).toMatchSnapshot();
        })

        it('render MovieCard component toMatchSnapshot', async () => {

            window.scrollTo = jest.fn();

            const { container } = render(<MovieCard movie={movie} onMovieChosen={onMovieChosen}/>);
            await waitFor(() => {fireEvent.click(container.querySelectorAll('a')[0])});
            await waitFor(() => {fireEvent.click(container.querySelectorAll('a')[1])});
        })
})
