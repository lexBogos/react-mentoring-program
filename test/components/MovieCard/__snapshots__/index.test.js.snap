// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard component render MovieCard component toMatchSnapshot 1`] = `
"<div className=\\"movieCard\\">
  <img src={[undefined]} width=\\"240\\" height=\\"320\\" />
  <div>
    <h3>
      Goodfellas
    </h3>
    <span />
  </div>
  <span className=\\"movieYear\\">
    1990
  </span>
  <div className=\\"movieCardActions\\">
    <a href=\\"#\\" onClick={[Function: onClick]}>
      edit
    </a>
    <a href=\\"#\\" onClick={[Function: onClick]}>
      delete
    </a>
    <a href=\\"#\\" onClick={[Function: onClick]}>
      ...
    </a>
  </div>
</div>"
`;
